batcher/aligned-batcher/src/lib.rs

```diff
@@ -1274,30 +1274,48 @@ impl Batcher {
             {
                 warn!("Failed to send task status to telemetry: {:?}", e);
             }
-            for entry in finalized_batch.into_iter() {
-                if let Some(ws_sink) = entry.messaging_sink {
-                    let merkle_root = hex::encode(batch_merkle_tree.root);
-                    send_message(
-                        ws_sink.clone(),
-                        SubmitProofResponseMessage::CreateNewTaskError(
-                            merkle_root,
-                            format!("{:?}", e),
-                        ),
-                    )
-                    .await
-                } else {
-                    warn!("Websocket sink was found empty. This should only happen in tests");
+
+            // decide if i want to flush the queue:
+            match e {
+                BatcherError::TransactionSendError(
+                    TransactionSendError::SubmissionInsufficientBalance,
+                )
+                | BatcherError::TransactionSendError(TransactionSendError::Generic(_)) => {
+                    self.flush_queue_and_clear_nonce_cache().await;
+                    self.send_close_messages(finalized_batch, batch_merkle_tree, &e)
```

Comment:
The `flush_queue_and_clear_nonce_cache` method is already sending a `BatchReset` message and since we haven't remove the finalized proofs from the queue we are sending two messages to them.